### ISSUE TITLE
Desync when alt-tabbing on sender side.

### ISSUE THREAD

**Issue by ItsVinnyX** [fzwoch/obs-teleport#41]
If I ALT-TAB on the sender side while in-game. the receiver will be super behind the current gameplay. Any fix for this?

Both my machines are connected via a 1 Gigabit Cat 6 cable.

**Comment by fzwoch** [comment]
I think that scenario is close to #37?

TL;DR is it seems that when the source OBS is blocking processing (loading a huge image, game capture things it seems) there will be a time of data starvation that is then caught up after the block. The result on the receiver seems to be the same - the timing gets delayed on the starvation part, but data is never discarded, re-synced after. Unless you hide/unhide the Teleport source.

**Comment by ItsVinnyX** [comment]
> I think that scenario is close to #37?
> 
> TL;DR is it seems that when the source OBS is blocking processing (loading a huge image, game capture things it seems) there will be a time of data starvation that is then caught up after the block. The result on the receiver seems to be the same - the timing gets delayed on the starvation part, but data is never discarded, re-synced after. Unless you hide/unhide the Teleport source.

So to fix it, I just need to unhide/hide teleport?

**Comment by fzwoch** [comment]
*InsertPerhapsMemeHere* - did you try?

**Comment by ItsVinnyX** [comment]
> _InsertPerhapsMemeHere_ - did you try?

Would it be on the sender or receiver?

**Comment by ItsVinnyX** [comment]
I do not think that worked. Is there a good setup to do this?

I use WiFi on both machines. Great speeds. Both machines are connected to each other by a Cat 6 Gigabit cable.
The only issue I faced with NDI was audio syncing.

**Comment by ItsVinnyX** [comment]
> That's indeed the scenario I had in #37, I specifically had on the list:
> 
> > What causes the lag doesn't matter, in another scenario it was triggered by a Game Capture source that is lagged for a second when tabbing into and out of a game.
> 
> But that was fixed with Teleport 0.5.0 at least for my test scenarios, so...
> 
> * Are you sure you are on the latest version of Teleport?
> * Are you using a filter to send the Teleport stream or just the base stream (as configured from the OBS "Tools" main menu)?
> * Where does the desync occur? In your audio monitoring device or live audio device? In a stream sent out from OBS? In a recording?
> 
> NDI audio sync issues was exactly what brought me here (to use Teleport instead), unfortunately they don't come from NDI itself but from the OBS implmentation of it and/or how OBS works in general. That means any other OBS based solution will always suffer from similar problems. Teleport with the latest update has build a bit around the limitations of OBS in this regard and is now more stable than the NDI plugin, but it's still not 100%, at least audio monitoring still gets horrible desyncs and also the filter based solution was so bad that I stopped trying to use it entirely.

But that was fixed with Teleport 0.5.0 at least for my test scenarios, so...

**Are you sure you are on the latest version of Teleport?** 
I am pretty sure I am. Just downloaded and replaced the .dll's

**Are you using a filter to send the Teleport stream or just the base stream (as configured from the OBS "Tools" main menu)?**

Sender:
![image](https://user-images.githubusercontent.com/5962854/179724915-c2194f31-ba35-4b86-b866-9e33b815985d.png)
![image](https://user-images.githubusercontent.com/5962854/179724706-84eb6a66-9fda-4c90-867f-ed99c15c1ae9.png)

Receiver:
![image](https://user-images.githubusercontent.com/5962854/179724840-bc81b1bb-af3f-4e44-9d3d-d7f10cd32927.png)
![image](https://user-images.githubusercontent.com/5962854/179724984-15aa7eb1-0253-4dd4-aaf8-3588f075bc3d.png)

**Where does the desync occur? In your audio monitoring device or live audio device? In a stream sent out from OBS? In a recording?**

Occurs on the Sender side. The preview and audio are delayed.

**Comment by fzwoch** [comment]
> Occurs on the Sender side. The preview and audio are delayed.

Can you elaborate on that one? I don't think Teleport should have any impact on the preview on the sender side.

**Comment by ItsVinnyX** [comment]
> > Occurs on the Sender side. The preview and audio are delayed.
> 
> Can you elaborate on that one? I don't think Teleport should have any impact on the preview on the sender side.
> 
> If you talk about preview and monitoring - I would not trust that. If the recording on that machine is in sync then OBS audio monitoring is at fault again. I can't tell if that is intended behavior or not.

Okay so to be honest I noticed it more when I ALT-TABBED but in general I am noticing some delays in audio and footage. Not on the sender but the receiver it desyncs on. Sorry about that.

**Comment by fzwoch** [comment]
And again, is that in the preview or final recording on the receiver?

**Comment by ItsVinnyX** [comment]
> And again, is that in the preview or final recording on the receiver?

It seems like it’s the preview. I use it to stream. I haven’t seen what it looks like on stream. But via audio monitor and visuals it’s definitely out of sync.

**Comment by fzwoch** [comment]
> @fzwoch You're talking about hiding/unhiding the Teleport source on the receiver? Then yes, this matches with my observations. You can actually hear it: if you hide/unhide too fast, then the audio will never stop. You need to hide so long (for me a little more than a second I'd say) until you no longer hear the audio, then unhide and audio is in sync again.

Oh God.. my sanity..

But if the recording is fine AV sync wise I think it is the same as #37. I don't really know what to do when the same data can be correct and incorrect at the same time. It may be that the audio meters reflect the potential audio monitoring path which may break up sync in favor of monitoring instead of delaying the audio for syncing it to the video.

**Comment by fzwoch** [comment]
Closing, as it seems to be the the same issue as the other.
